packages/botonic-plugin-nlu/src/scripts/retrain.js

```diff
@@ -0,0 +1,185 @@
+}
+
+async function askIfCorrect(intentsResp) {
+  const questions = [
+    {
+      type: 'confirm',
+      name: 'res',
+      message: `PREDICTED: [${intentsResp.intent}], CONFIDENCE: [${
+        intentsResp.confidence
+      }]. Is it correct?`
+    }
+  ]
+  return inquirer.prompt(questions)
+}
+
+async function askIfNewFile() {
+  const questions = [
+    {
+      type: 'confirm',
+      name: 'res',
+      message: `Do you want to create a new file for this intent?`
+    }
+  ]
+  return inquirer.prompt(questions)
+}
+
+async function askIfWantsToAdd() {
+  const questions = [
+    {
+      type: 'confirm',
+      name: 'res',
+      message: `Append to current intents?`
+    }
+  ]
+  return inquirer.prompt(questions)
+}
+
+async function askForCorrectIntent(intentsResp) {
+  let sortedIntents = intentsResp.intents.sort((a, b) => a.prob < b.prob)
+  let choices = sortedIntents.map(e => `${e.intent}, ${e.prob}`)
+  const questions = [
+    {
+      type: 'list',
+      name: 'res',
+      message: 'Which is the correct intent?',
+      choices: [...choices, 'NONE']
+    }
+  ]
+  return inquirer.prompt(questions)
+}
+
+async function askForFileName() {
+  const questions = [
+    {
+      name: 'name',
+      type: 'input',
+      message: 'What will be the name of the new intent?'
+    }
+  ]
+  return inquirer.prompt(questions)
+}
+
+async function askForLang(langs) {
+  let choices = langs.map(e => `${e}`)
+  const questions = [
+    {
+      type: 'list',
+      name: 'res',
+      message: 'Which is the language of this intent?',
+      choices: [...choices, 'New Language']
+    }
+  ]
+  return inquirer.prompt(questions)
+}
+
+async function askForNewLang() {
+  const questions = [
+    {
+      name: 'res',
+      type: 'input',
+      message:
+        'Enter the code of the new language (must be in ISO 639-3).\n\tFor details: https://iso639-3.sil.org/code_tables/639/data\n'
+    }
+  ]
+  return inquirer.prompt(questions)
+}
+
+async function retrain() {
+  let flagLang = process.argv.slice(3)[0]
+  let projectPath = process.env.INIT_CWD
+  let nluPath = path.join(projectPath, NLU_PATH)
+  let options = JSON.parse(
+    fs.readFileSync(path.join(projectPath, NLU_CONFIG_PATH))
+  )
+  if (flagLang) {
+    options = options.filter(config => config.LANG === flagLang)
+  }
+  let intentsPath = path.join(nluPath, INTENTS_DIRNAME)
+
+  let nlu = await new NLU(options)
+  while (true) {
+    let input = await askForUserInput()
+    if (input.res == 'exit()') {
+      break
+    } else {
+      let intents = await nlu.getIntents(input.res)
+      let isCorrect = await askIfCorrect(intents)
+      if (!isCorrect.res) {
+        let correctIntent = await askForCorrectIntent(intents)
+        if (correctIntent.res == 'NONE') {
+          let createNewFile = await askIfNewFile()
+          if (createNewFile.res) {
+            let newIntentPath = undefined
+            let langSelected = (await askForLang(fs.readdirSync(intentsPath)))
```

Comment:
usar funcio readdir?

Reply:
changed, also created a new fileUtils.js with all the fs utilities there. Additionally, I have seen during the refactor that they have the method mkdirSync(path) which I have used instead that mkdirp. Thanks to this I can remove one unnecessary dep.